libraries/AP_HAL_PX4/RCOutput.cpp

```diff
@@ -399,34 +395,12 @@ void PX4RCOutput::read_last_sent(uint16_t* period_us, uint8_t len)
     }
 }
 
-/*
-  update actuator armed state
- */
-void PX4RCOutput::_arm_actuators(bool arm)
-{
-    if (_armed.armed == arm) {
-        // already armed;
-        return;
-    }
-
-	_armed.timestamp = hrt_absolute_time();
-    _armed.armed = arm;
-    if (arm) {
-        // this latches ready_to_arm to true once set once. Otherwise
-        // we have a race condition with px4io safety switch update
-        _armed.ready_to_arm = true;
-    }
-    _armed.lockdown = false;
-    _armed.force_failsafe = false;
-}
-
 void PX4RCOutput::_send_outputs(void)
 {
     uint32_t now = AP_HAL::micros();
 
     if ((_enabled_channels & ((1U<<_servo_count)-1)) == 0) {
         // no channels enabled
-        _arm_actuators(false);
```

Comment:
safety switch and arm state are orthogonal. You can be armed with safety on or off.
See https://github.com/ArduPilot/ardupilot/pull/6104 for a related fix